server/users.go

```diff
@@ -1,317 +1,153 @@
-		Roles:       req.Roles,
-	}
-
-	res, err := store.Add(ctx, user)
-	if err != nil {
-		Error(w, http.StatusBadRequest, err.Error(), h.Logger)
-		return
-	}
-
-	if err != nil {
-		Error(w, http.StatusBadRequest, err.Error(), h.Logger)
-		return
-	}
-
-	su := newUserResponse(srcID, res.Name).WithPermissions(res.Permissions)
-	if _, hasRoles := h.hasRoles(ctx, ts); hasRoles {
-		su.WithRoles(srcID, res.Roles)
-	}
-	w.Header().Add("Location", su.Links.Self)
-	encodeJSON(w, http.StatusCreated, su, h.Logger)
+type userRequest struct {
+	ID       uint64 `json:"id"`
+	Name     string `json:"name"`
+	Provider string `json:"provider,omitempty"`
+	Scheme   string `json:"scheme,omitempty"`
 }
 
-// SourceUsers retrieves all users from source.
-func (h *Service) SourceUsers(w http.ResponseWriter, r *http.Request) {
-	ctx := r.Context()
-	srcID, ts, err := h.sourcesSeries(ctx, w, r)
-	if err != nil {
-		return
+func (r *userRequest) ValidCreate() error {
+	if r.Name == "" {
+		return errors.New("Name required on Chronograf User request body")
 	}
-
-	store := ts.Users(ctx)
-	users, err := store.All(ctx)
-	if err != nil {
-		Error(w, http.StatusBadRequest, err.Error(), h.Logger)
-		return
+	if r.Provider == "" {
+		return errors.New("Provider required on Chronograf User request body")
 	}
-
-	_, hasRoles := h.hasRoles(ctx, ts)
-	ur := make([]userResponse, len(users))
-	for i, u := range users {
-		usr := newUserResponse(srcID, u.Name).WithPermissions(u.Permissions)
-		if hasRoles {
-			usr.WithRoles(srcID, u.Roles)
-		}
-		ur[i] = *usr
+	if r.Scheme == "" {
+		return errors.New("Scheme required on Chronograf User request body")
 	}
+	return nil
+}
 
-	res := usersResponse{
-		Users: ur,
+func (r *userRequest) ValidUpdate() error {
+	if r.Name == "" && r.Provider == "" && r.Scheme == "" {
+		return errors.New("No fields to update")
 	}
-
-	encodeJSON(w, http.StatusOK, res, h.Logger)
+	return nil
 }
 
-// SourceUserID retrieves a user with ID from store.
-func (h *Service) SourceUserID(w http.ResponseWriter, r *http.Request) {
-	ctx := r.Context()
-	uid := httprouter.GetParamFromContext(ctx, "uid")
-
-	srcID, ts, err := h.sourcesSeries(ctx, w, r)
-	if err != nil {
-		return
-	}
-	store := ts.Users(ctx)
-	u, err := store.Get(ctx, uid)
-	if err != nil {
-		Error(w, http.StatusBadRequest, err.Error(), h.Logger)
-		return
-	}
+type userResponse struct {
+	Links    selfLinks `json:"links"`
+	ID       uint64    `json:"id"`
+	Name     string    `json:"name,omitempty"`
+	Provider string    `json:"provider,omitempty"`
+	Scheme   string    `json:"scheme,omitempty"`
+}
 
-	res := newUserResponse(srcID, u.Name).WithPermissions(u.Permissions)
-	if _, hasRoles := h.hasRoles(ctx, ts); hasRoles {
-		res.WithRoles(srcID, u.Roles)
+func newUserResponse(u *chronograf.User) *userResponse {
+	return &userResponse{
+		ID:       u.ID,
+		Name:     u.Name,
+		Provider: u.Provider,
+		Scheme:   u.Scheme,
+		Links: selfLinks{
+			Self: fmt.Sprintf("/chronograf/v1/users/%v", u.ID),
```

Comment:
Should not be %v
Should be the specific type if you know it (you do in this case)
Should be %d or if you change id to string %s